accepted/future-releases/nnbd/feature-specification.md

```diff
@@ -495,10 +498,26 @@ of a function call produces a result of type `Never`.
 
 The static type of a `throw e` expression is `Never`.
 
-In legacy mode, an override of operator== with no explicit parameter type
+An expression `e` of the form `e1 == e2` is treated as
+`let v1 = e1, v2 = e2 in v1 != null ? (v2 != null ? v1.==(v2) : false) : (v2 !=
+null ? false : true)`,
+where `v1.==(v2)` denotes an invocation of the instance method `operator ==`
+on `v1`, and the `let` construct is defined in the section 'Null aware
+operator'.
+
+_In particular, even if the static type of `e1` is potentially nullable,
```

Comment:
This disagrees with what I've written above about the types of the `Object` members when invoked on a nullable type, and I'm not sure I see a need to make an exception here.

Reply:
I think there is no real conflict here: Evaluation of `e1 == e2` is not an instance member invocation, and there is no way we can get away with ignoring the difference. So the rules about instance member invocations do not apply directly for `e1 == e2` anyway.

In particular, `Object.==` has a formal parameter of type `Object`, so we can't explain why `x == null` is allowed unless we make it explicit that this is different from an invocation of `operator ==` on the value of `x` (hence "assignable to `S?`", not ".. `S`").

Using **NonNull**(`T1`) as the receiver type when we look up the formal parameter type of `operator ==` is also justified by the special null pre-processing in the semantics.

Reply:
There may be no technical conflict, but it's confusing.  I think with the additional commentary it's fine though, thanks!